packages/address/src/address.ts

```diff
@@ -131,4 +133,32 @@ export default class Address extends AbstractAddress {
   static fromAddressOrString(address: string | AbstractAddress): AbstractAddress {
     return typeof address === 'string' ? this.fromString(address) : address;
   }
+
+  /**
+   * Takes an optional string and returns back an Address. If `address` is nilsy, return back random Address
+   *
+   * @param addressId - Can be a string containing Bech32, B256, or Public Key, or null/undefined
+   * @throws {Error}
+   * thrown if the input string is not nilsy and cannot be resolved to a valid address format
+   * @returns a new `Address` instance
+   */
+  static fromDynamicInput(addressId?: string | null): Address {
+    if (!addressId) {
+      return Address.fromRandom();
+    }
+
+    if (isPublicKey(addressId)) {
+      return Address.fromPublicKey(addressId);
+    }
+
+    if (isBech32(addressId)) {
+      return new Address(addressId as Bech32Address);
+    }
+
+    if (isB256(addressId)) {
+      return Address.fromB256(addressId);
+    }
+
+    throw new Error('Unknown address format: only Bech32, B256, or Public Key (512) supported');
+  }
 }
```

Comment:
Your logic is correct I think, it's good that we're flexible enough to accept empty input, but also make sure user won't use a wrong addresses

One thing I would change is:
The user shouldn't really need to dig into specific static methods (`fromDynamicInput`, `fromPublicKey`, etc..) to create an address.

instead of having static methods for doing it, I think `Address` constructor can centralize handling its inputs.

all the user would need is:

```
const address = new Address(dynamicInput);
```


if done as so, all this entrance static methods will become useless and can be deprecated.

Reply:
Agree on everything; I think we can leave it as is.

Reply:
I think for the dynamic input, we should not have the random method. This can be the source of issues, as the user didn't define anything. It should throw an error in this case. Other than that, it looks good.
```
if (!addressId) {
      return Address.fromRandom();
}
```

Reply:
We can aid this discussion by ensuring everyone's on the same page regarding a mantra: explicit is better than implicit. This philosophy is really good for building secure software.

With that in mind and contrasting what's being discussed here, I agree with @camsjams's point of having `new Address(userInput)` blow up if it's malformed, but if the user wants and knows that it can be malformed, using another, lower-level construct for that end. This is a solid approach that prevents any surprises (surprises are bad here!)

That said, I also agree with @luizstacio; `Address.fromDynamicInput()` (with no param) returning a random address is a relatively surprising behavior; as a user, I would immediately look for something named "random" like `Address.fromRandom()` if I was trying to create a random address. Allowing `Address.fromDynamicInput()` to return a random could cause more trouble than otherwise.

Reply:
I disagree.

you can check whole math/BN package, where we accept anything possible as input.
we use it a lot in front end and it's a really good DX... flexible to use, always work

```
bn('100');
bn(100);
bn('0x64');
bn(byteArray);
bn(buffer);
// same result for all prev examples...


// them you can do operations also with any input
bn(100).add('0x64').sub('50').add(bytesArray).format({precision: 3});
```

now imagine if for all of prev situations, I needed to:
```
bn.fromString('100');
bn.fromNumber(100);
bn.fromHex('0x64');
bn.fromByteArray(byteArray);
bn.fromBuffer(buffer)

// then for operations??
bn.fromString('100').addHex('0x64').subString('50');
```

to me DX and flexibility is key. if it's a valid input we should just keep it simple and accept. I see no reasons to block `new Address(userInput)` 
it's like saying:
"I know your input is valid, but please go and lose your time searching for the correct method".
Having multiple methods just makes the user think and search on things he doesn't really need to. 

I'm ok with blocking invalid/empty number, making it explicit for this case `Address.fromRandom()`



this is my opinion but of course feel free to dismiss my review and merge anyway if rest of the team wants on the other way

Reply:
I think bn makes sense because is fell on being a number that start with 0 and not random number what will be like Number() instead of returning 0 returning any random number between 0 and Infinity.

Reply:
I'm don't agree with empty validation as well, but I understand and am ok... what I'm not ok is with constructor accepting only `bech32`... why is that?

if we know it's a vlaid `bech32` or `b256` or `publicKey`, constructor should just accept

Reply:
Context is king. Within the `bn` example you (@LuizAsFight) gave, it does make sense to be a _lot_ more flexible. 

But this is a matter of security and safety; DX does _not_ take precedence over that. Explicitness is unarguably safer than implicitness; easy to back this up; tons of security research is out there. 

Having to explicitly and manually find the right `.from*()` method forces the user to think about how they're handling _public keys_. It goes without saying how sensitive this is.

If the user wants something dynamic, they must _opt-in_ by explicitly and manually picking `.fromDynamicInput()`. 

This removal of implicitness while manipulating _public keys_ is essential and security best practice.

Reply:
awesome thanks for context @digorithm , in terms of getting better security for critical operations, no harm on being more strict.